tally_ho/apps/tally/forms/site_info_form.py

```diff
@@ -0,0 +1,46 @@
+
+from django.forms import (
+    ModelForm,
+    TextInput
+)
+from django.conf import settings
+from django.contrib.sites.models import Site
+
+from tally_ho.apps.tally.models.site_info import SiteInfo
+
+
+class SiteInfoForm(ModelForm):
+    class Meta:
+        model = SiteInfo
+        fields = ['user_idle_timeout']
+
+        widgets = {
+            'user_idle_timeout': TextInput(attrs={'size': 50})
+        }
+
+    def __init__(self, *args, **kwargs):
+        super(SiteInfoForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        if self.is_valid():
+            cleaned_data = super(SiteInfoForm, self).clean()
+
+            return cleaned_data
+
+    def save(self):
+        if self.is_valid():
+            user_idle_timeout = self.cleaned_data.get('user_idle_timeout')
+            site_info = None
+
+            site_info_count = SiteInfo.objects.count()
```

Comment:
I think we should consider using `.exists()` here i.e. `SiteInfo.objects.exists()`  which is imho more suited for this kind of thing than `.count()` - i.e. for checking if a queryset returns anything.

But I think that doing that may not be what we want here.  If I am not wrong, you are trying to get the SiteInfo object - which is expected to be just one in the database, right?  If yes then I think it is safer to use [convenience helpers from django.contrib.sites](https://docs.djangoproject.com/en/3.0/ref/contrib/sites/#hooking-into-the-current-site-from-views) that allow us to get the current Site object, and then use that Site object to get its attached SiteInfo object. i.e you would do `site.siteinfo` to get the attached SiteInfo object.

I think you don't want to do `SiteInfo.objects.all().first()` because it is not future-proof i.e. if there ever comes a time we have more than one Site object in the DB, you are not guaranteed to get the correct SiteInfo object.  It would be better to either use the convenience methods mentioned above, or to explicitly get the right object using the SITE_ID setting e.g. `SiteInfo.objects.filter(site__id=settings.SITE_ID).first()`

Reply:
Nice makes sense.